OpenSim/Simulation/Model/Muscle.cpp

```diff
@@ -633,6 +633,26 @@ double Muscle::calcMusclePower(const SimTK::State& s) const
     return -getLengtheningSpeed(s) * getMuscleDynamicsInfo(s).tendonForce;
 }
 
+double Muscle::calcMuscleStiffness(const SimTK::State& s) const
+{
+    const MuscleDynamicsInfo& mdi = getMuscleDynamicsInfo(s);
+
+    const double dFmAT_dlceAT = mdi.fiberStiffnessAlongTendon;
+    const double dFt_dtl      = mdi.tendonStiffness;
+
+    double Ke = 0.;
+    if (!get_ignore_tendon_compliance()) {
+        // Compute the stiffness of the whole musculotendon actuator.
+        if (abs(dFmAT_dlceAT * dFt_dtl) > 0.0 &&
+            abs(dFmAT_dlceAT + dFt_dtl) > SimTK::SignificantReal) {
+            Ke = (dFmAT_dlceAT * dFt_dtl) / (dFmAT_dlceAT + dFt_dtl);
+        }
```

Comment:
This if-statement makes muscle stiffness calculations in `DeGrooteFregly2016Muscle` non-smooth. `DeGrooteFregly2016Muscle` is designed to be smooth and continuous everywhere, primarily for use in Moco. While muscle stiffness optimization isn't very common by Moco users, this changes the design intent for `DeGrooteFregly2016Muscle` and could lead to bad convergence if someone were to optimize muscle stiffness.

Reply:
Ah, thanks for clarifying.
I thought it was meant to catch zero-divided-by-zero or cases like that.
But overloading sounds good.